jwala-services/src/main/java/com/cerner/jwala/service/app/impl/ApplicationServiceImpl.java

```diff
@@ -132,18 +132,20 @@ public Application getApplication(final String name) {
     @Override
     public Application updateApplication(UpdateApplicationRequest updateApplicationRequest, User anUpdatingUser) {
         updateApplicationRequest.validate();
-
+        Application oldApplication = applicationPersistenceService.getApplication(updateApplicationRequest.getId());
         final Application application = applicationPersistenceService.updateApplication(updateApplicationRequest);
-        updateApplicationWarMetaData(updateApplicationRequest, application);
+        updateApplicationWarMetaData(updateApplicationRequest, application, oldApplication);
         return application;
     }
 
-    private void updateApplicationWarMetaData(UpdateApplicationRequest updateApplicationRequest, Application application) {
+    private void updateApplicationWarMetaData(UpdateApplicationRequest updateApplicationRequest, Application
+            application, Application oldApplication) {
         final String appWarName = application.getWarName();
         if (!StringUtils.isEmpty(appWarName)) {
             final String appName = application.getName();
             try {
-                String originalJsonMetaData = groupPersistenceService.getGroupAppResourceTemplateMetaData(application.getGroup().getName(), appWarName);
+                String originalJsonMetaData = groupPersistenceService.getGroupAppResourceTemplateMetaData(oldApplication.getGroup
+                        ().getName(), appWarName);
```

Comment:
I didn't think it was an issue because at this point the resource already exists, and the entity info was only used to upload the resource. I thought we didn't use the entity info to deploy the war, but I could be wrong ...